lib/rspec/core/runner.rb

```diff
@@ -84,8 +84,15 @@ def initialize(options, configuration=RSpec.configuration, world=RSpec.world)
       # @param out [IO] output stream
       def run(err, out)
         setup(err, out)
-        run_specs(@world.ordered_example_groups).tap do
-          persist_example_statuses
+
+        if RSpec.world.wants_to_quit
+          @configuration.reporter.report(0) do
+            @configuration.failure_exit_code
+          end
+        else
```

Comment:
We'd prefer a `return if RSpec.world.wants_to_quit` wants to quit here if possible, but understand the need to print a message, what does `return @configuration.failure_exit_code if RSpec.world.wants_to_quit` look like?

Reply:
That was my first approach, but I would I would get only the bare load error, but not all the rest of the candy RSpec adds to error messages. I decided that it's not a good user experience to print a differently formatted error just for this case.

Reply:
So, in the spirit of getting this over the line, could we refactor this to:

```
return @configuration.reporter.exit_early(@configuration.failure_exit_code) if RSpec.world.wants_to_quit
```

and then add a new public method on the reporter doing an exit pass?

Reply:
Ok, I'll do that change as well!